app/create_file.py

```diff
@@ -1 +1,56 @@
-# write your code here
+from datetime import datetime
+from sys import argv
+from os import mkdir, chdir
+
+filename = ""
+directories = []
+
+if "-f" in argv:
+    index = 0
+
+    for _ in range(len(argv) - 1):
+        index += 1
+        if argv[_: _ + 1] == ["-f"]:
+            break
+
+    filename = argv[index]
+
+    if "-d" in argv:
+        index2 = 0
+
+        for _ in range(len(argv) - 1):
+            index2 += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index2:index - 1]
+else:
+    filename = "file.txt"
+
+    if "-d" in argv:
+        index = 0
+
+        for _ in range(len(argv) - 1):
+            index += 1
+            if argv[_: _ + 1] == ["-d"]:
+                break
+
+        directories = argv[index:len(argv)]
```

Comment:
Logic error - you don't have default filename, it must be provided by user with `-f` flag, so you should create file only if `-f` with filename exists at argv.
You have repetitive code at 11-14 and 30-36 lines. Also that code is complicated - `argv` is a list and have method `index` which returns an index of the passed element or raise `ValueError` if the element not present in a list.
So, `argv[_: _ + 1]`, ` argv[index2:index - 1]`, `argv[index:len(argv)]` etc. can be changed to `argv.index("-param")`